src/lib/InstantSearch.js

```diff
@@ -195,7 +211,9 @@ class InstantSearch extends EventEmitter {
         typeof widget.dispose !== 'function'
       ) {
         throw new Error(
-          'The widget you tried to remove does not implement the dispose method, therefore it is not possible to remove this widget'
+          `The \`dispose\` method is required to remove the widget.
```

Comment:
Is there a reason why we do not enforce a dispose method, upfront, on init?

Reply:
I guess that's because at the time `dispose` didn't exist and we didn't want to introduce breaking changes for this.

Reply:
that makes sense. Thanks

Reply:
I think the reasoning indeed was that; combined with the fact that most people never `removeWidgets`, so the error would be useless at init